src/tool_operate.c

```diff
@@ -1866,9 +1870,9 @@ static CURLcode operate_do(struct GlobalConfig *global,
 #ifdef __AMIGA__
         if(!result && outs.s_isreg && outs.filename) {
           /* Set the url (up to 80 chars) as comment for the file */
-          if(strlen(url) > 78)
-            url[79] = '\0';
-          SetComment(outs.filename, url);
+          if(strlen(urlnode->url) > 78)
+            urlnode->url[79] = '\0';
+          SetComment(outs.filename, urlnode->url);
```

Comment:
Thinking of this code path: shouldn't this rather be done if the `--xattr` option is used, like it works for other platforms? That code path also scrubs off any username + password that might exist in the URL, while this amiga-specific block doesn't.

(No, this isn't really related to this particular change but it struck me now when I saw that this code is updated which means it is used...)

Reply:
I agree, although I can't see where the code needs moving to, unless it is literally a few lines up under the `/* Set file extended attributes */` comment?

Reply:
That's what I had in mind, yes. But I'm sorry, that's a separate issue and should be done separately from this PR. I just happened to notice since you fixed that code now.